jenkins/bootstrap.py

```diff
@@ -284,6 +284,7 @@ def start(gsutil, paths, stamp, node_name, version, repos):
     # Upload a link to the build path in the directory
     if paths.pr_build_link:
         gsutil.upload_text(paths.pr_build_link, paths.pr_path)
```

Comment:
Add a keyword argument to upload_text for optional headers instead-- this is a generally useful utility, and it wouldn't require two invocations.